src/deploy/functions/release/executor.ts

```diff
@@ -5,15 +5,32 @@ import { ThrottlerOptions } from "../../../throttler/throttler";
  * An Executor runs lambdas (which may be async).
  */
 export interface Executor {
-  run<T>(func: () => Promise<T>): Promise<T>;
+  run<T>(func: () => Promise<T>, opts?: RunOptions): Promise<T>;
```

Comment:
super nit (don't feel obligated to take it, and don't feel the need to go through re-review if you do). Typically callbacks are the last parameter. In langauges like Scala, Swift, and Ruby, this allows a special syntax where the callback is after the function call. We don't have that candy (yet) in JavaScript, but it's still pretty common.

Reply:
Ah interesting, definitely something to keep in mind — unfortunately in this case, `opts` is an optional parameter and `func` is required so I can't switch the parameter order without making `opts` required as well, but I'm guessing that the idiomatic approach would be to make `opts` required?

Reply:
The way we did this in cf3v2 is using overloading:

```
run<T>(func: () => Promise<T>): Promise<T>;
run<T>(opts: RuntimeOptions, func: () => Promise<T>): Promise<T>;

// impl
run<T>(funcOrOpts: RuntimeOptions | () => Promise<T>, func?: () => Promise<T>) {
  let opts: RuntimeOptions = {};
  if (func) {
    opts = funcOrOpts;
   } else {
     func = funcOrOpts;
   }
   // ...
}
```

If this were a public tool, we might push for this. But it's an internal tool that's used only a few times in our codebase, so please don't go through the trouble.